client/hostvolumemanager/host_volumes.go

```diff
@@ -7,20 +7,40 @@ import (
 	"context"
 	"errors"
 	"path/filepath"
+	"sync"
 	"time"
 
 	"github.com/hashicorp/go-hclog"
 	"github.com/hashicorp/go-multierror"
 	cstructs "github.com/hashicorp/nomad/client/structs"
 	"github.com/hashicorp/nomad/helper"
+	"github.com/hashicorp/nomad/nomad/structs"
 )
 
 var (
 	ErrPluginNotExists     = errors.New("no such plugin")
 	ErrPluginNotExecutable = errors.New("plugin not executable")
+	ErrUpdateTimeout       = errors.New("timeout updating client node")
 )
 
-type HostVolumeStateManager interface {
+type VolumeMap map[string]*structs.ClientHostVolumeConfig
+
+// UpdateVolumeMap returns true if it changes the volume map.
+func UpdateVolumeMap(volumes VolumeMap, name string, vol *structs.ClientHostVolumeConfig) (changed bool) {
```

Comment:
It doesn't look like we ever use the return value of this function.

Also, we're already holding the lock here or are working on a copy, right? We should probably document this in the docstring.

Reply:
we're working on a copy or a lock'd map. I figured the docstring saying it might change the volume map implied that, but I'll make it more explicit!

here are its usages:
* copy of config, uses `changed` here: https://github.com/hashicorp/nomad/pull/24613/files#diff-cf4c8c728ec5eb37deb3d780d75c3ca5dff1edf4b02fc46551fce35482510598R53
* ditto here: https://github.com/hashicorp/nomad/pull/24613/files#diff-cf4c8c728ec5eb37deb3d780d75c3ca5dff1edf4b02fc46551fce35482510598R145
* and this is a locked map, which doesn't care about `changed` because it's the first run on client start: https://github.com/hashicorp/nomad/pull/24613/files#diff-cf4c8c728ec5eb37deb3d780d75c3ca5dff1edf4b02fc46551fce35482510598R401

part of my goal for having this `UpdateVolumeMap` func was so that this list is a little easier to track down. the other plugin managers all have small variations of the same logic strewn about, so it was really hard to understand how it all connected! with this, I can set a breakpoint in this one func, and easily catch when this operation takes place.

Reply:
Ah, that caller in `batchFirstFingerprints` could probably be:

```
hostVolChanged = hvm.UpdateVolumeMap(newConfig.Node.HostVolumes, name, vol)
```

Reply:
true! I like little tidyings.